java/src/jmri/jmrit/beantable/oblock/OBlockTableModel.java

```diff
@@ -396,6 +396,11 @@ public void setValueAt(Object value, int row, int col) {
         if (!_tabbed && (super.getRowCount() == row)) { // editing tempRow
             switch (col) {
                 case SYSNAMECOL:
+                    if (!_manager.isValidSystemNameFormat((String) value)) {
+                        JOptionPane.showMessageDialog(null, Bundle.getMessage("BadNameOBlock"),
+                                Bundle.getMessage("ErrorTitle"), JOptionPane.WARNING_MESSAGE);
+                        return;
+                    }
```

Comment:
@dsand47 
Can you check this? Some managers accept a system name that doesn't have the correct prefix and then adds the prefix. What's the desired behavior here?

Reply:
"OB" is the only permitted prefix for OBlocks. Without checking the name here calling createNewOBlock with a bad name returns null. But returning null means an OBlock of that name already exists.

See the createBlock() method in jmri.jmrit.display.controlPanelEditor.EditCircuitFrame.java for the same. 

Doing it this way is much more informative to the user.